R/compute_centiloid.R

```diff
@@ -0,0 +1,181 @@
+#' are not specified, this function uses pre-defined slope and intercept based on
+#' the user's selections of tracer, pipeline, and reference region.
+#
+#' The following combinations of tracer, pipeline, and reference region are supported:
+#'
+#'  - `tracer = "18F-Florbetapir"`, `pipeline = "AVID FBP SUVR PIPELINE"`,
+#'    `ref_region = "Whole Cerebellum"`
+#'  - `tracer = "18F-Florbetapir"`, `pipeline = "BERKELEY FBP SUVR PIPELINE"`,
+#'    `ref_region = "Whole Cerebellum"`
+#'  - `tracer = "18F-Florbetaben"`, `pipeline = "BERKELEY FBB SUVR PIPELINE"`,
+#'    `ref_region = "Whole Cerebellum"`
+#'
+#' The equation used for the conversion is based on the following references:
+# nolint start
+#'  1. https://doi.org/10.1016/j.jalz.2018.06.1353 for 18F-Florbetapir, AVID FBP SUVR PIPELINE,
+#'     Whole Cerebellum.
+# nolint end
+#'  2. ADNI UCBerkeley Amyloid PET methods for 18F-Florbetapir, BERKELEY FBP SUVR PIPELINE,
+#'     Whole Cerebellum.
+#'  3. ADNI UCBerkeley Amyloid PET methods for 18F-Florbetaben, BERKELEY FBB SUVR PIPELINE,
+#'     Whole Cerebellum.
+#'
+#' Alternatively, the user can override the pre-selection by specifying `custom_slope`
+#' and `custom_intercept` instead. If `custom_slope` and `custom_intercept` are specified,
+#' `tracer`, `pipeline` and `ref_region` are ignored.
+#'
+#' If a matching combination of tracer, pipeline and reference region is not specified and
+#' `custom_slope` and `custom_intercept` are not specified, a warning is issued and
+#' `NA_real_` is returned.
+#'
+#' @return A numeric centiloid value
+#'
+#' @keywords com_bds_findings
+#' @family com_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' # Using standard parameters
+#' compute_centiloid(
+#'   tracer = "18F-Florbetapir",
+#'   pipeline = "AVID FBP SUVR PIPELINE",
+#'   ref_region = "Whole Cerebellum",
+#'   suvr = 1.25
+#' )
+#'
+#' # Using custom parameters
+#' compute_centiloid(
+#'   tracer = "MyTracer",
+#'   pipeline = "MyPipeline",
+#'   ref_region = "MyRegion",
+#'   suvr = 1.25,
+#'   custom_slope = 193,
+#'   custom_intercept = -187
+#' )
+compute_centiloid <- function(
+    tracer,
+    pipeline,
+    ref_region,
+    suvr,
+    custom_slope = NULL,
+    custom_intercept = NULL) {
+  valid_combinations <- tibble::tribble(
+    ~tracer,              ~pipeline,                     ~ref_region,          ~slope,   ~intercept,
+    "18F-Florbetapir",    "AVID FBP SUVR PIPELINE",      "Whole Cerebellum",   183,      -177,
+    "18F-Florbetapir",    "BERKELEY FBP SUVR PIPELINE",  "Whole Cerebellum",   188.22,   -189.16,
+    "18F-Florbetaben",    "BERKELEY FBB SUVR PIPELINE",  "Whole Cerebellum",   157.15,   -151.87
+  )
+
+  # Check custom_slope and custom_intercept
+  has_custom_slope <- !is.null(custom_slope)
+  has_custom_intercept <- !is.null(custom_intercept)
+
+  if (has_custom_slope != has_custom_intercept) {
+    cli::cli_abort("Both {.code custom_slope} and {.code custom_intercept}
```

Comment:
Across the actual function text, our convention in admiral is not to use `::` but instead add the functions to the imported list in `admiralneuro-package.R`